### PR TITLE
New resource 'azuread_group_member'

### PR DESCRIPTION
**Pull request by tiwood**
This adds a new resource to manage Azure AD Group Membership with Terraform.

- [x] New resource 'azuread_group_member'
- [x] Add tests
- [x] Add documentation

## Example Usage

```hcl

data "azuread_user" "my_user" {
  user_principal_name = "johndoe@hashicorp.com"
}

resource "azuread_group" "my_group" {
  name = "my_group"
}

resource "azuread_group_member" "test" {
  group_object_id   = "${azuread_group.my_group.id}"
  member_object_id  = "${data.azuread_user.my_user.id}"
}
```

## Argument Reference

The following arguments are supported:

* `group_object_id` - (Required) The Object ID of the Azure AD Group you want to add the Member to.
* `member_object_id` - (Required) The Object ID of the Azure AD Object you want to add as a Member to the Group. Supported Object types are Users, Groups or Service Principals.

### CODE REVIEW

**Comment by tiwood** [comment]
Related: #36

**Comment by tiwood** [comment]
@tombuildsstuff @katbyte 
any idea when this can be reviewed?
Thanks!

**Comment by katbyte** [comment]
@tiwood, we are currently busy with .12 and some work on the azurerm provider. Unfortunately I can't make any promises but I hope to earmark some time towards azuread in a couple weeks near the end of the month.

**Comment by katbyte** [comment]
Additionally i'd like to support setting group members via the group resource itself at the same time this is released.

**Comment by tiwood** [comment]
I will look into this in the next days.

File: azuread/resource_group_member.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	tenantID := client.TenantID
+
+	memberGraphURL := fmt.Sprintf("https://graph.windows.net/%s/directoryObjects/%s", tenantID, memberID)
+
+	properties := graphrbac.GroupAddMemberParameters{
+		URL: &memberGraphURL,
+	}
+
+	_, err := client.AddMember(ctx, groupID, properties)
+	if err != nil {
```
Comment (katbyte) [inline azuread/resource_group_member.go]: _minor_ these two lines can be merged:
```
	if _, err := client.AddMember(ctx, groupID, properties); err != nil {
```

File: azuread/resource_group_member.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+			if *servicePrincipal.ObjectID == memberID {
+				memberObjectID = *servicePrincipal.ObjectID
+				// we successfully found the directory object we're looking for, we can stop looping
+				// through the results
+				break
+			}
+		}
+
+		err = members.NextWithContext(ctx)
+		if err != nil {
```
Comment (katbyte) [inline azuread/resource_group_member.go]: These two lines could be merged

File: azuread/resource_group_member.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+			return fmt.Errorf("Error listing Azure AD Group Members: %s", err)
+		}
+	}
+
+	if memberObjectID == "" {
+		log.Printf("[DEBUG] Azure AD Group Member was not found (groupObjectId:%q / memberObjectId:%q ) - removing from state!", groupID, memberID)
```
Comment (katbyte) [inline azuread/resource_group_member.go]: Is this debug log redundant as we immediately return an error?

File: website/docs/r/group_member.markdown [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  group_object_id   = "${azuread_group.my_group.id}"
+  member_object_id  = "${data.azuread_user.my_user.id}"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `group_object_id` - (Required) The Object ID of the Azure AD Group you want to add the Member to.
```
Comment (katbyte) [inline website/docs/r/group_member.markdown]: Can we note that this forces a new resource?
```suggestion
* `group_object_id` - (Required) The Object ID of the Azure AD Group you want to add the Member to. Changing this forces a new resource to be created.
```

File: website/docs/r/group_member.markdown [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+* `member_object_id` - (Required) The Object ID of the Azure AD Object you want to add as a Member to the Group. Supported Object types are Users, Groups or Service Principals.
```
Comment (katbyte) [inline website/docs/r/group_member.markdown]:
```suggestion
* `member_object_id` - (Required) The Object ID of the Azure AD Object you want to add as a Member to the Group. Supported Object types are Users, Groups or Service Principals. Changing this forces a new resource to be created.
```

File: azuread/resource_group_member.go [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+	id := strings.Split(d.Id(), "/")
+	if len(id) != 2 {
+		return fmt.Errorf("ID should be in the format {groupObjectId}/{memberObjectId} - but got %q", d.Id())
+	}
+
+	groupID := id[0]
+	memberID := id[1]
+
+	resp, err := client.RemoveMember(ctx, groupID, memberID)
+
```
Comment (katbyte) [inline azuread/resource_group_member.go]:
```suggestion
```

**Comment by landro** [comment]
Any idea about ETA for this PR, @tiwood ? Can I help in any way?

**Comment by katbyte** [comment]
@landro, looks like PR comments still need to be addressed as well as adding the members property to the group resource

**Comment by evenh** [comment]
Would you be interested in a new PR (based on this PR) that fixes @katbyte's comments?

**Comment by katbyte** [comment]
@evenh, yep given this has sat for a month i think its reasonable to open a new one.

**Comment by landro** [comment]
Ok Great. You can expect a pull request from @evenh tomorrow.

**Comment by katbyte** [comment]
Looks like i had some fixes on a local branch, i'll push that now.

One thing also is i was going to use `{group_id}/member/{object_id}` to prevent conflicts with owners

**Comment by tiwood** [comment]
@katbyte @evenh @landro Thank you for helping, I'm currently pretty busy with customer projects and was unable to work on this in a timely matter.

As this is a much requested feature, your help is much appreciated.

**Comment by katbyte** [comment]
no bother @tiwood, thank you for starting them 🙂

**Comment by katbyte** [comment]
closing in favour of #100

**Pull request closed without merging**